go/lib.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright GLIDE-for-Redis Project Contributors - SPDX Identifier: Apache-2.0
+ */
+
+#include <stdarg.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdlib.h>
+
+typedef enum RequestErrorType {
+  Unspecified = 0,
+  ExecAbort = 1,
+  Timeout = 2,
+  Disconnect = 3,
+} RequestErrorType;
+
+/**
+ * The connection response.
+ *
+ * It contains either a connection or an error. It is represented as a struct instead of an enum for ease of use in the wrapper language.
+ *
+ * This struct should be freed using `free_connection_response` to avoid memory leaks.
+ */
+typedef struct ConnectionResponse {
+  const void *conn_ptr;
+  const char *error_message;
+  RequestErrorType error_type;
+} ConnectionResponse;
+
+/**
+ * Success callback that is called when a Redis command succeeds.
+ */
+typedef void (*SuccessCallback)(uintptr_t channel_address, const char *message);
```

Comment:
I'm not saying that it's a better solution, I just wonder whether the option was considered - I certainly didn't think about it back then.

Reply:
I talked with @acarbonetto about this and we don't have strong opinions on this change. It's more of a lateral move, so we ended up deciding it would be best to leave a TODO here and consider changing it later if preferred (tracked in our fork's backlog).